src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using StreamJsonRpc;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+{
+    internal class InProcLanguageServer
```

Comment:
why have an in-proc language server?  doc reason why we need this.

Reply:
Agreed on docs. Should we also rename this to something more like "LiveShareLanguageServer" so it's clear it's only being activated in the live share/remote case?

Reply:
(as opposed to the language server we have that runs OOP)

Reply:
Or maybe just a comment is better: we might want to dogfood the in-proc LSP client to at least see how our LSP experience is, and we'd be using this for that as well. I'll shut up now.